core/src/main/java/org/elasticsearch/common/util/concurrent/AbstractRunnable.java

```diff
@@ -36,7 +36,7 @@ public final void run() {
         try {
             doRun();
         } catch (InterruptedException ex) {
-            Thread.interrupted();
+            Thread.currentThread().interrupt();
```

Comment:
same hare - since we have on onFailure handler, calling is the equivalent of re-throwing the exception, imo.